src/vs/workbench/browser/parts/panel/panelPart.ts

```diff
@@ -181,7 +198,11 @@ export class PanelPart extends CompositePart<Panel> implements IPanelService {
 			}
 		}
 
-		return promise.then(() => this.openComposite(id, focus));
+		return promise.then(async () => {
+			const composite = await this.openComposite(id, focus);
```

Comment:
Why is this needed? Why can't we simply change the contextKey based on focus tracker only?